src/java.base/share/classes/sun/security/util/DisabledAlgorithmConstraints.java

```diff
@@ -159,6 +159,9 @@ public final boolean permits(Set<CryptoPrimitive> primitives,
         if (!cachedCheckAlgorithm(algorithm)) {
             return false;
         }
+        if (algorithm == null || algorithm.isEmpty()) {
```

Comment:
The check for null algorithm needs to happen before `cachedCheckAlgorithm` is called

Reply:
Fixed. Thanks for catching this!